lib/Runtime/Library/JavascriptString.h

```diff
@@ -312,6 +311,9 @@ namespace Js
         static bool IsASCII7BitChar(char16 ch) { return ch < 0x0080; }
         static char ToASCII7BitChar(char16 ch) { Assert(IsASCII7BitChar(ch)); return static_cast<char>(ch); }
 
+        // Share the string buffer, if both left and right strings are hosting the same content.
+        static void ShareStringBuffer(JavascriptString * leftString, JavascriptString * rightString);
```

Comment:
Why is this public? It looks to me like it should just be an internal detail of `JavascriptString`, so I was expecting it to be static.

Reply:
Right, public/protected stuff could be better.

Reply:
Oh; I see, it's called from `JavascriptStringHelpers` and not `JavascriptString` itself. Perhaps `JavascriptStringHelpers<T>` should be a friend of `JavascriptString`?